misc/ls-extensions/modules/metadata-to-record-generator/src/main/java/io/ballerina/generators/MetadataToRecordConverterFactory.java

```diff
@@ -0,0 +1,21 @@
+package io.ballerina.generators;
+
+import io.ballerina.generators.exceptions.MetadataToRecordGeneratorException;
+import io.ballerina.generators.platform.MetadataConverter;
+import io.ballerina.generators.platform.SalesforceMetadataConverter;
+
+import static io.ballerina.generators.MetadataToRecordConstants.SALESFORCE;
+import static io.ballerina.generators.MetadataToRecordConstants.UNSUPPORTED_METADATA_SOURCE;
+
+/**
+ * Factory class for the creation of metadata converter.
+ */
+public class MetadataToRecordConverterFactory {
+    public MetadataConverter getConvert(String schemaSource) throws MetadataToRecordGeneratorException {
```

Comment:
Let's add documentation comments describing the functionality of individual public functions